packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -160,52 +160,95 @@ - (UIImage *)extractIconFromData:(NSArray *)iconData
                                                        saturation:1.0
                                                        brightness:0.7
                                                             alpha:1.0]];
-  } else if ([iconData.firstObject isEqualToString:@"fromAsset"]) {
-    if (iconData.count == 2) {
-      image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]]];
-    } else {
-      image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]
-                                                   fromPackage:iconData[2]]];
-    }
   } else if ([iconData.firstObject isEqualToString:@"fromAssetImage"]) {
-    if (iconData.count == 3) {
-      image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]]];
-      id scaleParam = iconData[2];
-      image = [self scaleImage:image by:scaleParam];
-    } else {
-      NSString *error =
-          [NSString stringWithFormat:@"'fromAssetImage' should have exactly 3 arguments. Got: %lu",
-                                     (unsigned long)iconData.count];
-      NSException *exception = [NSException exceptionWithName:@"InvalidBitmapDescriptor"
-                                                       reason:error
-                                                     userInfo:nil];
-      @throw exception;
-    }
+    // Deprecated: This message handling for 'fromAssetImage' has been replaced by 'asset'.
+    // Refer to the flutter google_maps_flutter_platform_interface package for details.
+    NSAssert((iconData.count == 3), @"'fromAssetImage' should have exactly 3 arguments. Got: %lu",
+             (unsigned long)iconData.count);
+    image = [UIImage imageNamed:[registrar lookupKeyForAsset:iconData[1]]];
+    id scaleParam = iconData[2];
+    image = [self scaleImage:image by:scaleParam];
   } else if ([iconData[0] isEqualToString:@"fromBytes"]) {
-    if (iconData.count == 2) {
-      @try {
-        FlutterStandardTypedData *byteData = iconData[1];
+    // Deprecated: This message handling for 'fromBytes' has been replaced by 'bytes'.
+    // Refer to the flutter google_maps_flutter_platform_interface package for details.
+    NSAssert(iconData.count == 2,
+             @"'fromBytes' should have exactly 2 arguments, the bytes. Got: %lu",
+             (unsigned long)iconData.count);
+    @try {
+      FlutterStandardTypedData *byteData = iconData[1];
+      CGFloat screenScale = [[UIScreen mainScreen] scale];
```

Comment:
i could be wrong - is `mainScreen` deprecated?

Reply:
@hellohuanlin how to proceed with this?